include/realtime_tools/realtime_server_goal_handle.h

```diff
@@ -92,6 +95,8 @@ class RealtimeServerGoalHandle
   {
     if (req_execute_ && !req_succeed_ && !req_abort_ && !req_cancel_)
     {
+      std::lock_guard<std::mutex> guard(mutex_);
```

Comment:
If `runNonRealtime` is also locked with the same mutex, the `req_abort_` variable doesn't have to be an atomic. Same below.

Reply:
there's a check for req_abort_ outside the lock_guard, so it should be atomic

Reply:
in the `if` condition? I mean that's linked with 3(!!) `&&` which allows for all sorts of context switches in between. So I don't see how an atomic could help here or the condition doesn't make sense.

Reply:
fa6281a